tests/integration-tests/tests/budgets/test_budgets.py

```diff
@@ -0,0 +1,141 @@
+    # Update the cluster to add a budget
+    new_config_file = pcluster_config_reader(config_file="budget_with_notifications.yaml")
+    cluster.update(str(new_config_file), force_update="true")
+
+    # Try to describe the budget created by the cluster
+    budget_name_beginning = f"{cluster.name}-0-{region}"
+    budgets_description = budget_client.describe_budgets(AccountId=account_id)["Budgets"]
+    assert_that(
+        any([budget["BudgetName"].startswith(budget_name_beginning) for budget in budgets_description])
+    ).is_true()
+
+    # Update the cluster again with the initial file, removing the budget
+    cluster.update(str(init_config_file), force_update="true")
+
+    # Try to describe the budget again (should fail this time)
+    budgets_description = budget_client.describe_budgets(AccountId=account_id)["Budgets"]
+    assert_that(
+        any([budget["BudgetName"].startswith(budget_name_beginning) for budget in budgets_description])
+    ).is_false()
+
+
+def _update_cluster_and_check_fields(cluster, budget_client, account_id, region, pcluster_config_reader):
+    # Update cluster to have 3 budgets (one of each BudgetCategory)
+    add_multiple_budgets_config_file = pcluster_config_reader(config_file="all_budget_categories.yaml")
+    cluster.update(str(add_multiple_budgets_config_file), force_update="true")
+
+    # Update the cluster to change the budget properties
+    update_multiple_budgets_config_file = pcluster_config_reader(config_file="all_budget_categories_updated.yaml")
+    cluster.update(str(update_multiple_budgets_config_file), force_update="true")
+
+    # The budgets from the updated configuration
+    budget_list = cluster.config["DevSettings"]["Budgets"]
+    custom_budget, cluster_budget, queue_budget = budget_list[0], budget_list[1], budget_list[2]
+
+    budget_description = budget_client.describe_budgets(AccountId=account_id)["Budgets"]
+
+    # The expected budget names (without the hash), based on budget_builder.py
+    custom_budget_name_short = f"{cluster.name}-0-custom-{region}"
+    cluster_budget_name_short = f"{cluster.name}-1-{region}"
+    queue_budget_name_short = f"{cluster.name}-2-queue1-{region}"
+
+    # Look for the full names
+    for budget in budget_description:
+        name = budget["BudgetName"]
+        if name.startswith(custom_budget_name_short):
+            custom_budget_name = name
+        elif name.startswith(cluster_budget_name_short):
+            cluster_budget_name = name
+        elif name.startswith(queue_budget_name_short):
+            queue_budget_name = name
+
+    # Describe the updated budgets to verify that the properties were indeed updated
+    custom_budget_description = budget_client.describe_budget(
+        AccountId=account_id,
+        BudgetName=custom_budget_name,
+    )["Budget"]
+    cluster_budget_description = budget_client.describe_budget(
```

Comment:
["BudgetLimit"]["Amount"] is already included in `describe_budgets`, we don't need to get name from `describe_budgets` then get limit from `describe_budget`